src/scheduler/resv_info.c

```diff
@@ -231,6 +220,28 @@ query_reservations(server_info *sinfo, struct batch_status *resvs)
 			continue;
 		}
 
+		/* Make sure it is not a future reservation that is being deleted, if so ignore it */
+		if (resresv->resv->resv_state == RESV_BEING_DELETED && resresv->start > sinfo->server_time) {
+			schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_RESV, LOG_DEBUG,
+				resresv->name, "Future reservation is being deleted, ignoring this reservation");
+
+			sinfo->num_resvs--;
+			free_resource_resv(resresv);
+			cur_resv = cur_resv->next;
```

Comment:
Consider converting this while loop into a for loop so that cur_resv = cur_resv->next is part of the for loop header.

Reply:
Done